### DISCUSSION TITLE
How to limit the disk space usage of NonBlockingSoftIndexFileStore

### DISCUSSION THREAD

**Discussion by koshox** [infinispan/infinispan#10017] [Q&A] [answered]
I don't wanna NonBlockingSoftIndexFileStore saturate my hard disk , so how can I set a limit?
SingleFileStore has a param maxEntries, but I didn't find similar in NonBlockingSoftIndexFileStore.

**Comment by pruivo** [comment, marked as answer]
`NonBlockingSoftIndexFileStore` does not support it; it splits the data into multiple small files (you can specify the max size here, 16MB by default) but, when a file is full, a new one is created.
If you want (and understand the **risks**!), you can monitor the folder where the data is stored and remove the old files.

**Reply by koshox** [reply]
Thanks，if it's just data loss, I think it's acceptable, because it's used for caching

**Reply by pruivo** [reply]
Are you storing and removing entries from the cache manually? What is your configuration?

**Reply by koshox** [reply]
yes，I removing entries manually.
configuration:
```java
return new ConfigurationBuilder().clustering().cacheMode(CacheMode.INVALIDATION_ASYNC)
    .memory()
    .maxCount(2000)
    .persistence()
        .passivation(true)
        .addSoftIndexFileStore()
        .shared(false)
        .dataLocation("cache")
        .indexLocation("cache")
    .statistics()
    .enable()
    .build();
```

**Reply by koshox** [reply]
I want to keep 2000 cache entries in memory, use disk storage for the rest, persistence limit 20M disk or 20000 entries.
 Any suggestions please.

**Reply by pruivo** [reply]
I'll ask around. 
Currently, the only alternative is to use the `SingleFileStore`.

**Reply by pruivo** [reply]
actually, you can use expiration to remove the keys from in-memory and from the cache-store: https://infinispan.org/docs/stable/titles/configuring/configuring.html#expiration_configuring-memory-usage

**Reply by koshox** [reply]
Thank you，I'm having use lifespan. Will you remove SingleFileStore in 14.0?, I see SingleFileStoreConfiguration has been marked as deprecated.

**Reply by wburns** [reply]
SingleFileStore is not slated for removal in 14. The next possible removal will be in 15.

Also removing files manually will not work as you will get runtime exception due to the file missing that the index references.

I would suggest creating an issue against https://issues.redhat.com/projects/ISPN and we can track that there. I know there has been at least one other person who has shown desire for a feature like this.

**Reply by wburns** [reply]
I actually went ahead and created https://issues.redhat.com/browse/ISPN-13804 if you wanted to add any comments/vote/watch etc.

**Reply by koshox** [reply]
Ok, thanks.
